app/modules/annotations/parameters.py

```diff
@@ -23,13 +24,39 @@ class Meta(schemas.DetailedAnnotationSchema.Meta):
 
 class PatchAnnotationDetailsParameters(PatchJSONParameters):
     # pylint: disable=abstract-method,missing-docstring
-    OPERATION_CHOICES = (PatchJSONParameters.OP_REPLACE, PatchJSONParameters.OP_ADD)
+    OPERATION_CHOICES = (
+        PatchJSONParameters.OP_REPLACE,
+        PatchJSONParameters.OP_ADD,
+        PatchJSONParameters.OP_REMOVE,
+    )
 
-    PATH_CHOICES = tuple('/%s' % field for field in ('encounter_guid',))
+    PATH_CHOICES = tuple(
+        '/%s' % field
+        for field in (
+            'encounter_guid',
+            'keywords',
+        )
+    )
 
     @classmethod
     def add(cls, obj, field, value, state):
-        # Add and replace are the same operation so reuse the one method
+        if field == 'keywords':
+            from app.modules.keywords.models import Keyword
+
+            if isinstance(value, dict):  # (possible) new keyword
+                keyword = obj.add_new_keyword(
+                    value.get('value', None), value.get('source', None)
+                )
+                if keyword is None:
+                    return False
+            else:
+                keyword = Keyword.query.get(value)
+                if keyword is None:
+                    return False
+                obj.add_keyword(keyword)
+            return True
+
+        # otherwise, add and replace are the same operation so reuse the one method
```

Comment:
I feel like maybe there is some permission nuance here... you can write any annotation as a researcher currently, and it kinda needs to be that way for the initial creation. 

Once an annotation is on an encounter maybe at that point we should ensure the user has write permission for that encounter before we allow them to patch the annotation.

Reply:
PATCH on annotations (via resources.py) currently has all of this.  which i _think_ ends up resolving to what you are saying. ... **but** i am trying to confirm that now.
```    @api.permission_required(
        permissions.ObjectAccessPermission,
        kwargs_on_request=lambda kwargs: {
            'obj': kwargs['annotation'],
            'action': AccessOperation.WRITE,
        },
    )
    @api.login_required(oauth_scopes=['annotations:write'])
```

Reply:
ok, so it [looks like](https://github.com/WildMeOrg/houston/blob/dex205/app/modules/users/permissions/rules.py#L313) permissions is currently bubbling up to the **Asset** rather than Encounter.  for reasons this note says?  but i am wondering if we wanna keep it that way -- or **switch it to Encounter** if there indeed is one available.  :thinking: 

```            elif isinstance(self._obj, Annotation):
                # Annotation has no owner, but it has one asset, that has one asset_group that has an owner
                # (encounter is no longer required on Annotation, so best route to owner is via Asset/Group)
                has_permission = user == self._obj.asset.asset_group.owner
```

Reply:
I _think_ it would be slightly better if we added an 'if there is an encounter' fork to the permission check but it's probably fine through Asset